app/upgrades/v17/upgrades.go

```diff
@@ -0,0 +1,300 @@
+	for _, hostZone := range k.GetAllHostZone(ctx) {
+		for i, validator := range hostZone.Validators {
+			validator.SlashQueryInProgress = false
+			hostZone.Validators[i] = validator
+		}
+		k.SetHostZone(ctx, hostZone)
+	}
+}
+
+// Increases the community pool tax from 2 to 5%
+// This was from prop 223 which passed, but was deleted due to an ICS blacklist
+func IncreaseCommunityPoolTax(ctx sdk.Context, k distributionkeeper.Keeper) error {
+	params := k.GetParams(ctx)
+	params.CommunityTax = CommunityPoolTax
+	return k.SetParams(ctx, params)
+}
+
+// Updates the outer redemption rate bounds
+func UpdateRedemptionRateBounds(ctx sdk.Context, k stakeibckeeper.Keeper) {
+	for _, hostZone := range k.GetAllHostZone(ctx) {
+		// Give osmosis a bit more slack since OSMO stakers collect real yield
+		outerAdjustment := RedemptionRateOuterMaxAdjustment
+		if hostZone.ChainId == OsmosisChainId {
+			outerAdjustment = outerAdjustment.Add(OsmosisRedemptionRateBuffer)
+		}
+
+		outerMinDelta := hostZone.RedemptionRate.Mul(RedemptionRateOuterMinAdjustment)
+		innerMinDelta := hostZone.RedemptionRate.Mul(RedemptionRateInnerMinAdjustment)
+		innerMaxDelta := hostZone.RedemptionRate.Mul(RedemptionRateInnerMaxAdjustment)
+		outerMaxDelta := hostZone.RedemptionRate.Mul(outerAdjustment)
+
+		outerMin := hostZone.RedemptionRate.Sub(outerMinDelta)
+		innerMin := hostZone.RedemptionRate.Sub(innerMinDelta)
+		innerMax := hostZone.RedemptionRate.Add(innerMaxDelta)
+		outerMax := hostZone.RedemptionRate.Add(outerMaxDelta)
+
+		hostZone.MinRedemptionRate = outerMin
+		hostZone.MinInnerRedemptionRate = innerMin
+		hostZone.MaxInnerRedemptionRate = innerMax
+		hostZone.MaxRedemptionRate = outerMax
+
+		k.SetHostZone(ctx, hostZone)
+	}
+}
+
+// Update rate limits based on current TVL
+func UpdateRateLimitThresholds(ctx sdk.Context, sk stakeibckeeper.Keeper, rk ratelimitkeeper.Keeper) {
+	for _, rateLimit := range rk.GetAllRateLimits(ctx) {
+		stDenom := rateLimit.Path.Denom
+		hostDenom := stDenom[2:]
+
+		// Lookup the associated host zone to get the chain ID
+		hostZone, err := sk.GetHostZoneFromHostDenom(ctx, hostDenom)
+		if err != nil {
+			ctx.Logger().Error(fmt.Sprintf("host zone not found for denom %s", hostDenom))
+			continue
+		}
+
+		// Determine the expected rate limit threshold for the chain
+		updatedThreshold, ok := UpdatedRateLimits[hostZone.ChainId]
+		if !ok {
+			ctx.Logger().Error(fmt.Sprintf("rate limit not specified for %s", hostZone.ChainId))
+			continue
+		}
+
+		// If the expected threshold is 0, that means there should be no rate limit
+		// Remove the rate limit in this case
+		if updatedThreshold.IsZero() {
+			rk.RemoveRateLimit(ctx, rateLimit.Path.Denom, rateLimit.Path.ChannelId)
+			continue
+		}
+
+		rateLimit.Quota.MaxPercentRecv = updatedThreshold
+		rateLimit.Quota.MaxPercentSend = updatedThreshold
+		rk.SetRateLimit(ctx, rateLimit)
+	}
+}
+
+// Rate limits transfers to osmosis across each stToken
```

Comment:
Why only Osmosis and not any DEX?  People who are trying to exit a lot of liquidity could just use astroport or kujira fin or something instead right?

Reply:
Would we want different limits per DEX based on amount of liquidity in the pools or would these limits be general. Is this constraint actually something which is on the Osmosis side?

Reply:
We'll likely do a global rate limit for each sttoken in the future, across all channels

Reply:
Reopening this as I think it's a very good question. 

While we wait for the global rate limit across all channels, should we RL transfers to Neutron, Secret and Kujira too (main other DEXs)?

Reply:
we decided on doing just osmosis for now, marking this resolved